# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546903617 28800
#      Mon Jan 07 15:26:57 2019 -0800
# Branch stable
# Node ID 6aa84dc27d01ee47915adcfa53a42f0cd983d438
# Parent  d41c487e5de24cca423780dc67ffd84e411eadce
Fix memory leak in mkoctfile.

* mkoctfile.in.cc (tmp_objfile_name): Use delete to free memory created with
new.

diff --git a/src/mkoctfile.in.cc b/src/mkoctfile.in.cc
--- a/src/mkoctfile.in.cc
+++ b/src/mkoctfile.in.cc
@@ -598,20 +598,22 @@ tmp_objfile_name (void)
 
   char *ctmpl = new char [tmpl.length () + 1];
 
   ctmpl = strcpy (ctmpl, tmpl.c_str ());
 
   // mkostemps will open the file and return a file descriptor.  We
   // won't worry about closing it because we will need the file until we
   // are done and then the file will be closed when mkoctfile exits.
-
   octave_mkostemps_wrapper (ctmpl, 2);
 
-  return std::string (ctmpl);
+  std::string retval (ctmpl);  // make C++ string from filled-in template
+  delete [] ctmpl;
+
+  return retval;
 }
 
 static void
 clean_up_tmp_files (const std::list<std::string>& tmp_files)
 {
   for (const auto& file : tmp_files)
     octave_unlink_wrapper (file.c_str ());
 }
